From 59742b90b098b00157439770db7a70b0b502ecdf Mon Sep 17 00:00:00 2001
Message-Id: <59742b90b098b00157439770db7a70b0b502ecdf.1417465742.git.chang-joon.lee@intel.com>
In-Reply-To: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
References: <664d811e547fec51f0453c5acda92012de288ecd.1417465742.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Tue, 4 Nov 2014 16:21:45 +0530
Subject: [PATCH 06/11] SQUASHME! [VPG]: drm/i915: Correction in
 drm_mode_setplane lock handling

SQUASHME! - This patch should be squashed into the following existing patch:

Author: Vidya Srinivas <vidya.srinivas@intel.com>
Date:   Sat Jul 5 00:31:06 2014 +0530
Change-Id: Icd2fd0bf8890c6864b2b671e4560ef1f5438295b
        FOR_UPSTREAM [VPG]: drm/i915: Sprite flip Notification

Issue: GMIN-1455
Change-Id: I74c8cc46de38db2e26d75060fbe98ff9009daa6c
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
Signed-off-by: Chuansheng Liu <chuansheng.liu@intel.com>
Signed-off-by: Xiaoming Wang <xiaoming.wang@intel.com>
---
 drivers/gpu/drm/drm_crtc.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/drm_crtc.c b/drivers/gpu/drm/drm_crtc.c
index fd6346f..d065e6d 100644
--- a/drivers/gpu/drm/drm_crtc.c
+++ b/drivers/gpu/drm/drm_crtc.c
@@ -2307,7 +2307,7 @@ int drm_mode_setplane(struct drm_device *dev, void *data,
 		spin_lock_irqsave(&dev->event_lock, flags);
 		if (file_priv->event_space < sizeof(e->event)) {
 			spin_unlock_irqrestore(&dev->event_lock, flags);
-			goto out;
+			goto unlock;
 		}
 		file_priv->event_space -= sizeof(e->event);
 		spin_unlock_irqrestore(&dev->event_lock, flags);
@@ -2316,7 +2316,7 @@ int drm_mode_setplane(struct drm_device *dev, void *data,
 			spin_lock_irqsave(&dev->event_lock, flags);
 			file_priv->event_space += sizeof(e->event);
 			spin_unlock_irqrestore(&dev->event_lock, flags);
-			goto out;
+			goto unlock;
 		}
 
 		e->event.base.type = DRM_EVENT_FLIP_COMPLETE;
@@ -2345,6 +2345,8 @@ int drm_mode_setplane(struct drm_device *dev, void *data,
 		}
 		old_fb = NULL;
 	}
+
+unlock:
 	drm_modeset_unlock_all(dev);
 
 out:
-- 
1.7.9.5

